### ISSUE TITLE
Subquery in `ON` clause missing parentheses

### ISSUE THREAD

**Issue by bspot** [knex/knex#5046]
# Environment

Knex version: 0.21.19 up to the one in KNEX Query Lab
Database + version: MySql 8
OS: Linux

# Bug

When using a subquery in an `ON` clause, the subquery is not wrapped in parentheses, which does not work in MySQL.
I also can not use `as()`, because using `as tablename` is also not accepted by MySQL.

## Example 

https://michaelavila.com/knex-querylab/?dialect=mysql2&query=NYOwpgHgFA5AxgCwIYgOZhgSgHQBswBmALgFID2AliLAWWTADQAEUAVpkwLwB8Tr2ZajAoATRkxidxoSDTpZsAZzD44RWKIW4KAWwrqAjJmNA

```
knex('change').leftJoin('foo', (j) => j.on('id', '=', knex('foo').select('id').limit(1)))
```

## Resulting SQL:
```
select * from `change` left join `foo` on `id` = select `id` from `foo` limit 1;
```

This gives the following error:
```
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'select `id` from `foo` limit 1' at line 1
```

## Expected:
```
select * from `change` left join `foo` on `id` = (select `id` from `foo` limit 1);
```

**Comment by bspot** [comment]
Oh, that was fast. Thank you very much!

**Comment by kibertoad** [comment]
Released in 1.0.4
